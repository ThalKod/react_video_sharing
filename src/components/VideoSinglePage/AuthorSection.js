/* eslint-disable react/self-closing-comp */

import React from "react";
import { Link } from "react-router-dom";
import { request } from "utils";

import Avatar from "components/Header/Avatar";

export default class AuthorSection extends React.Component{

  state = {
    videosCount: 0,
    subscribersCount: 0,
    username: ""
  };

  componentDidMount = () => {
    const { author } = this.props;

    const getVideoCountPromise = request("get", `/user/${author}/video/count`);
    const getSubscribersCountPromise = request("get", `/user/${author}/subscribers/count`);
    const getUserNamePromise = request("get", `/user/${author}/name`);

    Promise.all([getSubscribersCountPromise, getVideoCountPromise, getUserNamePromise])
        .then(res => {
          const [{ data: { subscribersCount } }, { data: { videosCount }}, { data: { username }}] = res;
          this.setState({ subscribersCount, videosCount, username });
        })
        .catch(err => console.log(err));
  };


  render(){
    const { viewCount } = this.props;
    const { videosCount, subscribersCount, username } = this.state;

    return(
        <div className="author">
          <Link to="/"><Avatar username={username} className="sv-avatar"/></Link>
          <div className="sv-name">
            <div><Link to="/">{username.charAt(0).toUpperCase() + username.slice(1)}</Link>{` . ${videosCount} Videos`}</div>
            <div className="c-sub">
              <div className="c-f">
                Subscribe
              </div>
              <div className="c-s">
                {subscribersCount}
              </div>
              <div className="clearfix"></div>
            </div>
          </div>
          <div className="sv-views">
            <div className="sv-views-count">{ viewCount } Views</div>
          </div>
          <div className="clearfix"></div>
        </div>
    )
  }
}
