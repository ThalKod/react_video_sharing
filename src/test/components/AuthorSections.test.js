import React from "react";
import {Link} from "react-router-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json"

import AuthorSection from "../../components/VideoSingle/AuthorSection";

describe("AuthorSection", () => {

  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<AuthorSection/>);
  });

  it("should render the component correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render the correct state", () => {
    const state = {
      videosCount: 8,
      subscribersCount: 25,
      username: "Test Name"
    };

    wrapper.setState(state); // Setting the initial state...

    expect(
        wrapper.containsMatchingElement(
            <div><Link to="/">{state.username.charAt(0).toUpperCase() + state.username.slice(1)}</Link>{` . ${state.videosCount} Videos`}</div>
        )
    ).toBe(true);

    expect(
        wrapper.containsMatchingElement(
            <div className="c-s">
              {state.subscribersCount}
            </div>
        )
    ).toBe(true);

  })
});
