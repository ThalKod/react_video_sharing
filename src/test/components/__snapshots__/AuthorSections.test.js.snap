// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthorSection should render the component correctly 1`] = `
<div
  className="author"
>
  <Link
    replace={false}
    to="/channel/DSACQEK1238321JLKNE"
  >
    <_default
      className="sv-avatar"
      username=""
    />
  </Link>
  <div
    className="sv-name"
  >
    <div>
      <Link
        replace={false}
        to="/channel/DSACQEK1238321JLKNE"
      />
       . 0 Videos
    </div>
    <Connect(Subscriber)
      id="DSACQEK1238321JLKNE"
      sub={0}
    />
  </div>
  <div
    className="sv-views"
  >
    <div
      className="sv-views-count"
    >
      0
       Views
    </div>
  </div>
  <div
    className="clearfix"
  />
</div>
`;
